src/sentry/models/organization.py

```diff
@@ -120,6 +122,53 @@ def get_for_user(self, user, scope=None, only_visible=True):
             return [r.organization for r in results if scope in r.get_scopes()]
         return [r.organization for r in results]
 
+    def get_member_top_dog_organizations(self, user_id: int, queryset: QuerySet = None) -> QuerySet:
+        from sentry.models import OrganizationMember, OrganizationMemberTeam, Team
+
+        # get orgs and orgmemberIDs for the user
+        if queryset:  # queryset is a QuerySet of valid orgs
+            query = list(
+                OrganizationMember.objects.filter(
+                    organization__in=queryset, user_id=user_id
+                ).values_list("organization_id", "id")
+            )
+        else:
+            query = list(
+                OrganizationMember.objects.filter(user_id=user_id).values_list(
+                    "organization_id", "id"
+                )
+            )
+
+        organizations, org_members = zip(*query)
+        organizations = list(organizations)
+        org_members = list(org_members)
+
+        # get owner teams
+        owner_teams = list(
+            Team.objects.filter(
+                organization_id__in=organizations, org_role=roles.get_top_dog().id
+            ).values_list("id", flat=True)
+        )
+        # get owners from owner teams
+        owners = list(
+            OrganizationMemberTeam.objects.filter(
+                team_id__in=owner_teams,
+                organizationmember_id__in=org_members,
+            ).values_list("organizationmember_id", flat=True)
+        )
+        # get corresponding orgs from org list
+        orgs = {organizations[org_members.index(m)] for m in owners if m in org_members}
+
+        # get owners from orgs
+        owner_role_orgs = list(
+            OrganizationMember.objects.filter(
+                role=roles.get_top_dog().id, user_id=user_id
+            ).values_list("organization_id", flat=True)
+        )
+        orgs.update(owner_role_orgs)
```

Comment:
If this is a `get` method - why are we modifying the table??

Reply:
this is a set of org_ids

Reply:
It is a bit confusing at first glance. Maybe, instead of mutating the value of `orgs`, make `owner_role_orgs` a set instead of a list and then
```
return self.filter(id__in=orgs.union(owner_role_orgs), status=OrganizationStatus.ACTIVE)
```